breadth_first_search/BreadthFirstSearch.java

```diff
@@ -0,0 +1,65 @@
+import java.util.*;
+
+public class BreadthFirstSearch <T> {
+
+    // HasMap  of lists for Adjacency List Representation
+    public HashMap<Integer, ArrayList<T>> adj;
+
+    //Function to add an edge into the breadthFirstTraversal
+    private void addEdge(int source, T destination) {
```

Comment:
No, it is not. In a `HashMap`, any hashable object can act as an index.

The soul of the `adj` (adjacency list) is that it holds all the nodes that are connected to the given node in the graph. In this implementation, since nodes will be of type `T`, the `HashMap` will need to map `T` to `ArrayList<T>`.

This is the reason why you need to query for `T` type object from index in the traverse method.